src/t_hash.c

```diff
@@ -61,22 +62,24 @@ int hashTypeGetFromZiplist(robj *o, sds field,
 {
     unsigned char *zl, *fptr = NULL, *vptr = NULL;
     int ret;
+    listContainerType *lct = (o->encoding == OBJ_ENCODING_ZIPLIST) ?
+        &listContainerZiplist : &listContainerListpack;
 
-    serverAssert(o->encoding == OBJ_ENCODING_ZIPLIST);
+    serverAssert(o->encoding == OBJ_ENCODING_ZIPLIST || o->encoding == OBJ_ENCODING_LISTPACK);
 
     zl = o->ptr;
-    fptr = ziplistIndex(zl, ZIPLIST_HEAD);
+    fptr = lct->listIndex(zl, 0);
     if (fptr != NULL) {
-        fptr = ziplistFind(zl, fptr, (unsigned char*)field, sdslen(field), 1);
+        fptr = lct->listFind(zl, fptr, (unsigned char*)field, sdslen(field), 1);
```

Comment:
we need to conduct some benchmark to make sure we're not suffering a performance hit from that additional indirection

Reply:
I will do it.